src/dune_rpc_impl/server.ml

```diff
@@ -329,25 +335,33 @@ let handler (t : _ t Fdecl.t) handle : 'build_arg Dune_rpc_server.Handler.t =
     Handler.declare_notification rpc Procedures.Server_side.log
   in
   let () = Handler.implement_request rpc Procedures.Public.ping (fun _ -> Fiber.return) in
-  let () =
-    let build _session targets =
+  let implement_request_pending_action decl ~f =
```

Comment:
`implement`? Looking at the code from my understanding I think `register` would better describe what it does.